src/wp_ecc_kmgmt.c

```diff
@@ -1376,7 +1376,10 @@ static int wp_ecc_gen_set_template(wp_EccGenCtx* ctx, wp_Ecc* ecc)
         }
     }
     if (ok) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
         XSTRNCPY(ctx->curveName, name, sizeof(ctx->curveName));
```

Comment:
Change this to sizeof(...) - 1 and set the last character to '\0'

Reply:
oddly, I did not need this.